src/utils/input_parser.h

```diff
@@ -18,7 +18,7 @@ namespace io {
 
 struct CLArgs;
 
-Input parse(const CLArgs& cl_args);
+Input parse(std::string input_str, Servers& servers, ROUTER router, bool geometry);
```

Comment:
We should make `input_str` a `const std::string&` to avoid copying it. Also since we used to pass a `const CLArgs&`, we may as well have a `const Servers&`: `parse` should not change the server config and this is also what the `Input` ctor consumes down the line.

Reply:
Not relevant any more as signature has changed.